core/modules/replica/sql/replication.sql

```diff
@@ -0,0 +1,759 @@
+
+  CONSTRAINT `request_qserv_remove_replica_fk_1`
+    FOREIGN KEY (`request_id` )
+    REFERENCES `request` (`id` )
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+)
+ENGINE = InnoDB;
+
+-- -----------------------------------------------------
+-- Table `request_qserv_set_replicas`
+-- -----------------------------------------------------
+--
+-- Extended parameters of the 'QSERV:SET_REPLICAS' requests
+--
+DROP TABLE IF EXISTS `request_qserv_set_replicas` ;
+
+CREATE TABLE IF NOT EXISTS `request_qserv_set_replicas` (
+
+  `request_id`  VARCHAR(255) NOT NULL ,
+
+  --
+  -- space separated sequence of pairs representing databases
+  -- and chunks:
+  --
+  --   '<database1>:<chunk1> <database2>:<chunk2> ...'
+  --
+  `replicas` LONGTEXT NOT NULL ,
+  `force`    BOOLEAN  NOT NULL ,
+
+  CONSTRAINT `request_qserv_set_replicas_fk_1`
+    FOREIGN KEY (`request_id` )
+    REFERENCES `request` (`id` )
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+)
+ENGINE = InnoDB;
+
+
+
+-- -----------------------------------------------------
+-- Table `replica`
+-- -----------------------------------------------------
+--
+-- ATTENTION: replicas have dependencies onto worker and database
+--            entries of the Configuration. Removing workers or
+--            databases from the correspondig configuration tables
+--            will also eliminate replicas.
+--
+DROP TABLE IF EXISTS `replica` ;
+
+CREATE TABLE IF NOT EXISTS `replica` (
+
+  `id`  INT NOT NULL AUTO_INCREMENT ,
+
+  `worker`   VARCHAR(255) NOT NULL ,
+  `database` VARCHAR(255) NOT NULL ,
+  `chunk`    INT UNSIGNED NOT NULL ,
+
+  `verify_time` BIGINT UNSIGNED NOT NULL ,
+
+  PRIMARY KEY           (`id`) ,
+  KEY                   (`worker`,`database`) ,
+  UNIQUE  KEY `replica` (`worker`,`database`,`chunk`) ,
+
+
+  CONSTRAINT `replica_fk_1`
+    FOREIGN KEY (`worker` )
+    REFERENCES `config_worker` (`name` )
+    ON DELETE CASCADE
+    ON UPDATE CASCADE ,
+
+  CONSTRAINT `replica_fk_2`
+    FOREIGN KEY (`database` )
+    REFERENCES `config_database` (`database` )
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `replica_file`
+-- -----------------------------------------------------
+DROP TABLE IF EXISTS `replica_file` ;
+
+CREATE TABLE IF NOT EXISTS `replica_file` (
+
+  `replica_id`  INT NOT NULL ,
+
+  `name`  VARCHAR(255)    NOT NULL ,
+  `size`  BIGINT UNSIGNED NOT NULL ,
+  `mtime` INT    UNSIGNED NOT NULL,
+  `cs`    VARCHAR(255)    NOT NULL ,
+
+  `begin_create_time`  BIGINT UNSIGNED NOT NULL ,
+  `end_create_time`    BIGINT UNSIGNED NOT NULL ,
+
+  UNIQUE  KEY `file` (`replica_id`,`name`) ,
+
+  CONSTRAINT `replica_file_fk_1`
+    FOREIGN KEY (`replica_id` )
+    REFERENCES `replica` (`id` )
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+)
+ENGINE = InnoDB;
+
+
+SET SQL_MODE=@OLD_SQL_MODE ;
+SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS ;
+SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS ;
```

Comment:
General comment for the whole file - PRIMARY KEY needs to be defined on each table unless there is no obvious primary key. I see that some tables have it defined but many do not.

Reply:
Agreed. I will fix this.

Reply:
One more comment. Those missing PRIMARY KEY case are in tables where the key is mentioned in the FOREIGN KEY CONSTRAINT. I always though, this kind of constrain would automatically introduce a key. The only difference i that those won't be UNIQUE. So, performance wise this isn't an issue.

Reply:
True, for InnoDB mysql creates an index on FK (unless there is another suitable index). But it's not related to my PK question. If you do not specify PK then mysql can create additional "ROWID" column in your table and use it as PK. You probably do not want this most of the time.